src/main/scala/com/fulcrumgenomics/commons/io/CommandLineTool.scala

```diff
@@ -0,0 +1,216 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2019 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.fulcrumgenomics.commons.io
+
+import java.nio.file.Path
+
+import com.fulcrumgenomics.commons.util.LazyLogging
+import com.fulcrumgenomics.commons.CommonsDef._
+
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
+
+/** Provides methods for an executable to run it. For example, methods to run this
+  * executable using provided arguments, test if this executable is available, etc.
+  */
+trait CommandLineTool extends LazyLogging {
+
+  /** The name of the executable. */
+  val Executable: String
```

Comment:
```suggestion
  val executable: String
```
variables should be lower camel case, except for constants (which are part of objects): https://docs.scala-lang.org/style/naming-conventions.html